numpy/testing/nose_tools/utils.py

```diff
@@ -377,44 +377,42 @@ def assert_equal(actual, desired, err_msg='', verbose=True):
 
     # Inf/nan/negative zero handling
     try:
-        # If one of desired/actual is not finite, handle it specially here:
-        # check that both are nan if any is a nan, and test for equality
-        # otherwise
-        if not (gisfinite(desired) and gisfinite(actual)):
```

Comment:
I don't think there was any reason to special-case the infinite cases - those work fine under normal `==`